Automattic/pocket-casts-android
# Settings sync - Refactor appearance settings to be `UserSetting`s

## Description
This builds on #1227 by converting the Appearance settings to be `UserSetting`s.

## Testing Instructions

Note that there is a pre-existing bug [when switching between Plus and non-Plus themes on a Free account](https://github.com/Automattic/pocket-casts-android/issues/1228).

For each setting in the "Appearance" section:

1. Fresh install the app
2. Verify that the setting has the same default value that it had before
3. Change the setting to a new value
4. Force close the app
5. Verify that the setting has retained the new value
6. Verify that the app's behavior reflects the updated setting value.

## Checklist
